### ISSUE TITLE
Question about the poor performance on custom data

### ISSUE THREAD

**Issue by Uio96** [liuyuan-pal/Gen6D#29]
Thank you so much for this great work. I also tried configuring Gen6D and successfully applied it to the mouse data provided.

However, when I tried to apply Gen6D to my collected data, the results were not satisfying as the detector could not work properly.

There are some previous trials in https://github.com/liuyuan-pal/Gen6D/issues/4, https://github.com/liuyuan-pal/Gen6D/issues/11 and https://github.com/liuyuan-pal/Gen6D/issues/24. I noticed that you mentioned the issue of z flip or reference target being too small there, so I was careful with those factors. I also tried to put an ArUco tag as the background. I once thought my target object might be too hard, so I changed to a mouse like your demo one. But Gen6D still did not work on my own data.

A  sample is like this 
![0](https://user-images.githubusercontent.com/13082153/200211427-8c9f84c4-7739-4a95-a4bb-5f68a758cafe.jpg)
while my segmentation and labeling should be good 
<img src="https://user-images.githubusercontent.com/13082153/200212401-3ae28525-17b3-4662-a3f0-157dc1bedc6b.png " width="300">

Do you have any thoughts on the reasons for the failure cases? I wonder if I have to print out another ArUco tag (denser than the one I use right now) similar to the one used in your demo mouse video or https://github.com/liuyuan-pal/Gen6D/issues/24#issuecomment-1280417908. Have you tried capturing the registration video without the ArUco tag board?

Thanks a lot.

**Comment by EternalGoldenBraid** [comment]
Have I understood correctly that the pointcloud reconstruction is used to define the object center for each reference frame viewpoint? The center is then used obtain reference frames which are centered at the object?

**Comment by EternalGoldenBraid** [comment]
Have I understood correctly that the pointcloud reconstruction is used to label a 3D bounding box for defining the object center for each reference frame viewpoint? The 3D center is then projected to a 2D center for reference frames?

**Comment by liuyuan-pal** [comment]
Hi, for your problem, you may resize the input image a little bit smaller so that the detector is able to detect the region correctly because the detection seems to be too incorrect. Another problem is that you may still crop the mouse tighter so that the mouse in the reference image would look larger.

**Comment by liuyuan-pal** [comment]
> Have I understood correctly that the pointcloud reconstruction is used to label a 3D bounding box for defining the object center for each reference frame viewpoint? The 3D center is then projected to a 2D center for reference frames?

Yes, this is almost correct! I compute a 3D bounding (inscribed) sphere (instead of a 3D bounding box) for the object and the reference images will include the projection of the whole sphere.

**Comment by Uio96** [comment]
> Hi, for your problem, you may resize the query image a little bit smaller so that the detector is able to detect the region correctly because the detection seems to be too incorrect. Another problem is that you may still crop the mouse tighter so that the mouse in the reference image would look larger.

Thank you so much for the reply. Your suggestion did help me with my mouse case.

**Comment by Uio96** [comment]
I have a follow-up question. I tested on more cases and found that the detection result was very sensitive to the size of the target. For example, the following two inputs (both of them have been resized to a height of 640) may look similar as regards their target sizes, but the performance was quite different. 

1) sample 1
![0](https://user-images.githubusercontent.com/13082153/200463993-ed061f7f-c6bc-4217-84db-6b63c948489e.jpg)

2) sample 2
![0 (1)](https://user-images.githubusercontent.com/13082153/200464024-bf36b8f7-5961-4797-b528-0a5ea082f004.jpg)

I realized you did mention that 
> The detector and viewpoint selector will be used for the initialization of the poses, which assume the object in the query image looks relatively similar to the reference images in terms of scale and in-plane orientation (the given z-axis is the up direction).
https://github.com/liuyuan-pal/Gen6D/issues/4#issuecomment-1186131326

However, some of my registration images are quite close to the target (I would say it is more similar to sample 1)
<img src="https://user-images.githubusercontent.com/13082153/200464112-4b4b2e24-600a-439c-82af-922f3a1d0f67.jpg " width="100">

So my question is, when is Gen6D more guaranteed to succeed?

**Comment by liuyuan-pal** [comment]
Hi, to guarantee the success of the detector, we require the scale difference between the query image and the reference image is not too large. In the implementation, the reference image has a size of 128, so the detected bounding box in the query image is supposed to be 128/2=64 to 128*2=256. Scale differences larger than 2.0 or 0.5 would decrease the success rate. The reason is that we use convolutions on different scales to detect the object. The predefined scale range is 0.5 to 2.0.

**Comment by liuyuan-pal** [comment]
BTW, we will always crop and resize the reference image to 128*128 according to the `object_point_cloud.ply`, which is not affected by your registered image size.

**Comment by Shivam7Sharma** [comment]
Hi,

Thank you for this software. I did tight cropping of the ply file, and I reduced the size of your mouse in the readme by 0.7 and 0.8 scale, but the results are very poor. I am using your video in the readme for reference and testing. Is there something I am doing wrong? I followed all the steps in the custom object readme.
